app/models/user/UserService.scala

```diff
@@ -125,15 +145,13 @@ object UserService extends FoxImplicits {
     }
   }
 
-  def changePassword(user: User, newPassword: String)(implicit ctx: DBAccessContext) = {
-    if (user.isActive)
-      Mailer ! Send(DefaultMails.changePasswordMail(user.name, user.email))
-
-    UserDAO.changePassword(user._id, newPassword).map { result =>
-      UserCache.invalidateUser(user.id)
-      result
+  def changePasswordInfo(loginInfo:LoginInfo, passwordInfo:PasswordInfo) = {
+    for{
+      user <- findOneByEmail(loginInfo.providerKey)
+      _ <- UserDAO.changePasswordInfo(user._id, passwordInfo)(GlobalAccessContext)
+    }yield{
```

Comment:
generally (for the future: please insert spaces next to curly braces, no `}yield{` and no `={`)

Reply:
Ok